### PR TITLE
Replace subset_to_apply in docs where possible

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address references to `subset_to_apply` when called with `get_data` in the documentation.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/cubeviz/helper.py [hunk 1]
```diff
@@ -123,8 +123,8 @@ def specviz(self):
     def get_data(self, data_label=None, spatial_subset=None, spectral_subset=None, function=None,
                  cls=None):
         """
-        Returns data with name equal to data_label of type cls with subsets applied from
-        subset_to_apply.
+        Returns data with name equal to ``data_label`` of type ``cls`` with subsets applied from
+        ``spatial_subset`` and/or ``spectral_subset`` using ``function`` if applicable.
```
Comment (rosteen) [inline jdaviz/configs/cubeviz/helper.py:127]: Maybe say where this is applicable?

Comment (javerbukh) [reply jdaviz/configs/cubeviz/helper.py:127]: I expand on that more in the parameters section, is there something specifically you want to see here?

Comment (rosteen) [reply jdaviz/configs/cubeviz/helper.py:127]: Ah I see, missed it since it wasn't in the diff. This looks fine 👍

**Review by rosteen — APPROVED** [review]

**Review by kecnry — APPROVED** [review]
I set to backport just so the built docs on 3.5.x are updated.  Thanks!

**Pull request merged**